### PR TITLE
fix coredump caused by ERR_INCOMPLETE_DATA, which is caused by imcomp…

### PR DESCRIPTION
**Pull request by qinzuoyan**
…leted checkpointing during learning, refer to #5

### CODE REVIEW

File: src/replica_server/replication_lib/replica_init.cpp [hunk 1]
```diff
@@ -269,9 +269,8 @@ error_code replica::init_app_and_prepare_list(bool create_new)
                         _private_log->max_commit_on_disk(),
                         finish_time - start_time
                         );
+
                     _private_log->check_valid_start_offset(get_gpid(), _app->init_info().init_offset_in_private_log);
-                    set_inactive_state_transient(true);
```
Comment (imzhenyu) [inline src/replica_server/replication_lib/replica_init.cpp]: Delete this stmt will cause the initial transitions, e.g., ```PS_INACTIVE => PS_PRIMARY``` or ```PS_INACTIVE => PS_SECONDARY``` at startup failed.

Comment (qinzuoyan) [inline src/replica_server/replication_lib/replica_init.cpp]: but this stmt also appears in replica_stub.cpp:361, which will overwrite all old values, so it's unnecessary to set here.

Comment (imzhenyu) [inline src/replica_server/replication_lib/replica_init.cpp]: Here we set the value according to per-replica state, and in replica-stub we set that for shared state. It is best to leave both, similar to the case where we set last error multiple times in different scope for different scenarios.

**Comment by imzhenyu** [comment]
@qinzuoyan I will fix app-veyor test first before the merge.

**Comment by qinzuoyan** [comment]
@imzhenyu , I have re-commit to the pull request, and set_inactive_state_transient as following:
- in replica_init.cpp:275, set_inactive_state_transient(true) if open replica and replay private log succeed.
- in replica_stub.cpp:364, just leave inactive_state_transient as its old value.

Is it ok for you?

**Comment by imzhenyu** [comment]
Thx, it is good for me.

**Pull request merged**